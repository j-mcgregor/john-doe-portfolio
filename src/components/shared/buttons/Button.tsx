import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSpinner } from '@fortawesome/free-solid-svg-icons'

const Button: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement>> = ({
    type,
    value,
    onClick,
    disabled,
    className = '',
}) => (
    <button
        type={type}
        onClick={onClick}
        className={`btn ${className}`}
        data-testid={value}
        disabled={disabled}
    >
        {value}
        {disabled && <FontAwesomeIcon icon={faSpinner} spin className="ml1" />}
    </button>
)

export default Button
