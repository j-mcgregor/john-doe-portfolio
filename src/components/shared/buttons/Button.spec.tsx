import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Button from './Button'

describe('Button', () => {
    it('should render a button', () => {
        const mockFn = jest.fn()
        const component = render(
            <Button type="button" value="Submit" onClick={mockFn} />
        )

        expect(component.baseElement).toMatchSnapshot()
    })

    it('should trigger the function onClick', () => {
        const mockFn = jest.fn()
        const { getByText } = render(
            <Button type="button" value="Submit" onClick={mockFn} />
        )

        fireEvent.click(getByText('Submit'))

        expect(mockFn).toHaveBeenCalled()
    })
})
