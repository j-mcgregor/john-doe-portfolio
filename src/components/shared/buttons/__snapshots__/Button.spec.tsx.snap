// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button 1`] = `
<body>
  <div>
    <button
      class="btn "
      data-testid="Submit"
      type="button"
    >
      Submit
    </button>
  </div>
</body>
`;
